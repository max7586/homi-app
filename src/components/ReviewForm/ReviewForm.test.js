import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import ReviewForm from './ReviewForm';
import { BrowserRouter } from 'react-router-dom'

describe('ReviewForm component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <BrowserRouter>
        <ReviewForm />
      </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ReviewForm />)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });


});