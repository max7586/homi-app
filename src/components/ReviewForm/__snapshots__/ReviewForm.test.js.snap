// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewForm component renders the UI as expected 1`] = `
<form
  className="ReviewForm"
  onSubmit={[Function]}
>
  <div
    className="text"
  >
    <textarea
      aria-label="Type a review..."
      className="Textarea "
      cols="30"
      id="text"
      name="text"
      placeholder="Type a review or make an offer .."
      required={true}
      rows="3"
    />
  </div>
  <div
    className="select"
  >
    <label
      htmlFor="rating"
    >
      Rate this property!
    </label>
    <select
      aria-label="Rate this property!"
      id="rating"
      name="rating"
      required={true}
    >
      <option
        value="1"
      >
        1 Star
      </option>
      <option
        value="2"
      >
        2 Stars
      </option>
      <option
        value="3"
      >
        3 Stars
      </option>
      <option
        value="4"
      >
        4 Stars
      </option>
      <option
        value="5"
      >
        5 Stars
      </option>
    </select>
  </div>
  <button
    className="Button "
    type="submit"
  >
    Post review
  </button>
</form>
`;
